### DISCUSSION TITLE
Plugin help - Custom text/images on Monsters

### DISCUSSION THREAD

**Discussion by Michael-Koers** [runelite/runelite#17382] [Q&A] [answered]
Hi all, just started out with first my first RuneLite plugin for funsies. 

Still very much in the process of finding out about all the classes, services etc there are available to use. What I'm looking for is a way to display some custom text on or above NPC's. I've found the ```NpcOverlayService.java``` and ```ModelOutlineRenderer.java```, however as far as I can tell with my limited knowledge, these don't seem to support custom text. 

Is there currently a way to do this? Any pointers are helpful, be it to another plugin or just the name of the class.

Cheers!

**Comment by Nightfirecat** [comment, marked as answer]
`HighlightedNpc#isName()` indicates to the NPC Overlay whether the NPC's name should be drawn: https://github.com/runelite/runelite/blob/77ea5364da9cc2c491dd1c8c7b046759e4a0cd43/runelite-client/src/main/java/net/runelite/client/game/npcoverlay/HighlightedNpc.java#L50, https://github.com/runelite/runelite/blob/77ea5364da9cc2c491dd1c8c7b046759e4a0cd43/runelite-client/src/main/java/net/runelite/client/game/npcoverlay/NpcOverlay.java#L147-L156

**Reply by Michael-Koers** [reply]
Thanks, that helped! Found the Overlay class and extending from that in combination with the OverlayUtil got me to the next step.
